src/openrct2-ui/windows/Footpath.cpp

```diff
@@ -1057,26 +1057,24 @@ static void footpath_remove_tile_element(TileElement* tileElement)
 static TileElement* footpath_get_tile_element_to_remove()
 {
     TileElement* tileElement;
-    int32_t x, y, z, zLow;
+    int32_t z, zLow;
 
-    x = gFootpathConstructFromPosition.x / 32;
-    y = gFootpathConstructFromPosition.y / 32;
-    if (x >= 256 || y >= 256)
+    if (map_is_location_valid(gFootpathConstructFromPosition))
     {
         return nullptr;
     }
 
-    z = (gFootpathConstructFromPosition.z >> 3) & 0xFF;
-    zLow = z - 2;
+    z = gFootpathConstructFromPosition.z & std::numeric_limits<int32_t>::max();
+    zLow = z - (2 * COORDS_Z_STEP);
```

Comment:
This code can be a lot simpler if we find out the bitwise and is not necessary.

Reply:
The & is most likely due to dl being traditionally the register for base height. Its not required but the old code had the other feature of maxing out at 255 (smallZ) which we will need to replicate.